### ISSUE TITLE
bug(MatSlider): ERROR TypeError: Cannot read properties of undefined (reading '_hostElement')

### ISSUE THREAD

**Issue by ice909** [angular/components#29738]
### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### Description

```html
<mat-slider min="1" max="5" step="0.5" value="1.5"></mat-slider>
```
Using this MatSlider component will print the following error in the console
![image](https://github.com/user-attachments/assets/18944c3e-ec30-4eaa-9ade-a42f59db0144)

### Reproduction

StackBlitz link: https://material-14.angular.cn/guide/getting-started
Steps to reproduce:
1. Following the official example, the console reports this error after MatSlider is displayed on the page

### Expected Behavior

No error

### Actual Behavior

error

### Environment

- Angular: 18.2.4
- CDK/Material: 18.2.4
- Browser(s): Chrome 128.0.6613.84
- Operating System (e.g. Windows, macOS, Ubuntu): Linux Deepin v23

**Comment by crisbeto** [comment]
You're looking at the docs for a much older version than the one you're running. The correct docs are at https://material.angular.io/components/slider/overview. That being said, the error message we log can be improved. I've sent a fix for the error message at #29745.

**Comment by ice909** [comment]
> You're looking at the docs for a much older version than the one you're running. The correct docs are at https://material.angular.io/components/slider/overview. That being said, the error message we log can be improved. I've sent a fix for the error message at #29745.

Thank you！
